src/modules/trajectory_generation/domain/spline_generation/cubic_spline_generator.cpp

```diff
@@ -1,10 +1,160 @@
 #include <cubic_spline_generator.hpp>
+#include <core_datastructures/Posture.hpp>
+#include <vector>
+#include <cmath>
+#include <eigen3/Eigen/Dense>
+#include <iostream>
+
 
 namespace trajectory_generation{
-    CubicSplineGenerator::CubicSplineGenerator(const core_datastructures::Posture& start, const core_datastructures::Posture& goal) : start(start), goal(goal) {
```

Comment:
start and goal have already been initialized in this line. No need to assign the values again in the line below.